subsys/bluetooth/host/l2cap.c

```diff
@@ -1945,7 +1945,8 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 	if ((buf->len <= ch->tx.mps) &&
 	    (net_buf_headroom(buf) >= BT_L2CAP_BUF_SIZE(0))) {
 		LOG_DBG("len <= MPS, not allocating seg for %p", buf);
-		seg = net_buf_ref(buf);
+		/* move `buf` to `seg`. `buf` now borrows `seg`. */
+		seg = buf;
```

Comment:
After doing this this, wouldn't it be safer & clearer to set `buf = NULL;`? (and then modify the various `(buf == seg)` checks later in this function to `(buf == NULL)`?

Reply:
Yes. But this whole function is getting replaced in https://github.com/zephyrproject-rtos/zephyr/pull/72090. We could delay my change until that is in?

Reply:
> Yes. But this whole function is getting replaced in #72090. We could delay my change until that is in?

I'm really fine either way, as long as both changes go in before the 3.7 feature freeze.